pkg/analysis/optionalorrequired/testdata/src/c/c.go

```diff
@@ -0,0 +1,27 @@
+package c
+
+type OptionalOrRequiredTestStruct struct {
+	RequiredEnumField RequiredEnum // want "field RequiredEnumField must be marked as kubebuilder:validation:Optional or kubebuilder:validation:Required"
+
+	KubebuilderRequiredEnumField KubeBuilderRequiredEnum // want "field KubebuilderRequiredEnumField must be marked as kubebuilder:validation:Optional or kubebuilder:validation:Required"
+
+	OptionalEnumField OptionalEnum // want "field OptionalEnumField must be marked as kubebuilder:validation:Optional or kubebuilder:validation:Required"
+
+	KubebuilderOptionalEnumField KubeBuilderOptionalEnum // want "field KubebuilderOptionalEnumField must be marked as kubebuilder:validation:Optional or kubebuilder:validation:Required"
+}
+
+// +required
+// +kubebuilder:validation:Enum=Foo;Bar;Baz
+type RequiredEnum string // want "required should not be defined on a type RequiredEnum"
```

Comment:
To fit with other error messages from this rule, perhaps 
```suggestion
type RequiredEnum string // want "type RequiredEnum should not be marked as required"
```